Networking/NetworkingTests/Remote/ProductsRemoteTests.swift

```diff
@@ -95,4 +95,45 @@ class ProductsRemoteTests: XCTestCase {
 
         wait(for: [expectation], timeout: Constants.expectationTimeout)
     }
+
+    // MARK: - Search Products
+
+    /// Verifies that searchProducts properly parses the `products-load-all` sample response.
+    ///
+    func testSearchProductsProperlyReturnsParsedProducts() {
+        let remote = ProductsRemote(network: network)
+        let expectation = self.expectation(description: "Load All Products")
+
+        network.simulateResponse(requestUrlSuffix: "products", filename: "products-load-all")
```

Comment:
It might make sense to have a dedicated payload for this test. I would suggest performing a search on a real store, using an actual search term, check visually that the response contains the actual product expected, and then translate that into an actual test, otherwise I am not sure this test is really testing what it should. 

That being said, it is true that in these tests we are not testing what the API returns, but the parsers, so what I just said in the paragraph above might be a formalism, so 🤷‍♂

Reply:
yep I think it's worth testing with real response - even though most of the test is on the mapper we can still make sure the mapper works for the API response at the time of the development. Updated in the last 2 commits!